File: leetcode/3sum/first.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int> > output;
        vector<int> row;
        std::sort(nums.begin(), nums.end());
        set<vector<int>> myset;
        for(int z = 0; z < nums.size(); z++) {
            int left = z+1;
            int right = nums.size()-1;
            while(left<right) 
            {
                int value = nums[z] + nums[left] + nums[right];
                bool found = false;
                if(value == 0) 
                {
                    row.push_back(nums[z]);
                    row.push_back(nums[left]);
                    row.push_back(nums[right]);
                    myset.insert(row);
                    row.clear();
                    left++;
                    right--;
                }
                else if(value > 0) 
                {
                    right--;
                }
                else 
                {
                    left++;
                }
            }
        }
        for(auto iterator = myset.begin(); iterator!=myset.end(); iterator++) {
            output.push_back(*iterator);
        }
        return output;
    }
};

File: leetcode/3sumk/first.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        bestValue = False
        for x in range(len(nums)):
            left = x + 1
            right = len(nums)-1
            while left<right:
                val = nums[x] + nums[left] + nums[right]

                if bestValue is False:
                    bestValue = val
                    
                elif abs(target-bestValue) > abs(target-val):
                    bestValue = val
                    if target==val:
                        target

                if val > target:
                    right = right -1
                else:
                    left = left + 1
                    
        return bestValue
                